lisp/mastodon-notifications.el

```diff
@@ -0,0 +1,168 @@
+;;; mastodon-notifications.el --- Notification functions for mastodon.el
+
+;; Copyright (C) 2017 Johnson Denen
+;; Author: Johnson Denen <johnson.denen@gmail.com>
+;; Version: 0.5.4
+;; Homepage: https://github.com/jdenen/mastodon.el
+
+;; This file is not part of GNU Emacs.
+
+;; This file is part of mastodon.el.
+
+;; mastodon.el is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; mastodon.el is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with mastodon.el.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; mastodon-notification.el provides notification functions.
+
+;; todo: - integrate media
+;;       - extend notification types
+;;       - replace `You' with `your_display_name'(@`your_user_name')
+;;       - generalize mastodon-tl--more or rebind it in notifications
+;;         example
+;;       - fix thread access
+;;       - Add keybinding of mastodon-notifications--get to `N'
+
+;; ;; One approach for generalizing the get-more functionality
+;; (defun mastodon-tl--get-more (more-function id-fun goto-fun
+;;                               processing-function)
+;;   "can work for both oldert toots and newest toots"
+;;   (interactive)
+;;   (let* ((tl (mastodon-tl--timeline-name))
+;;       (id (funcall id-fun))
+;;       (json (funcall more-function tl id)))
+;;     (when json
+;;       (with-current-buffer (current-buffer)
+;;      (let ((inhibit-read-only t))
+;;        (goto-char (funcall goto-fun))
+;;        (funcall processing-function json))))))
+;;
+;; ;; Equivelent to the mastodon-tl--more
+;; (mastodon-tl--get-more 'mastodon-tl--more-json 'mastodon-tl--oldest-id
+;;                        point-max mastodon-tl--timeline)
+;;
+;; ;; Equivelent to the mastodon-tl--update
+;; (mastodon-tl--get-more 'mastodon-tl--update-json 'mastodon-tl--newest-id
+;;                        point-min mastodon-tl--timeline)
+
+
+;;; Code:
+
+;;(require 'mastodon-media)
+(require 'mastodon-http)
+(require 'mastodon-tl)
+
+(defgroup mastodon-notifications nil
+  "Mastodon Notifications."
+  :prefix "mastodon-notifications-"
+  :group 'mastodon)
+
+(defvar mastodon-notifications--types
```

Comment:
Why not have a plist of type to handling-function?  Then you don't repeat yourself below in the dispatching function.

Reply:
Fixed, turned into plist